src/entity/SquadPostModeration.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  Column,
+  CreateDateColumn,
+  Entity,
+  ManyToOne,
+  PrimaryColumn,
+  UpdateDateColumn,
+} from 'typeorm';
+import { Post, PostType } from './posts';
+import { Source } from './Source';
+
+export enum SquadPostModerationStatus {
+  Approved = 'approved',
+  Rejected = 'rejected',
+  Pending = 'pending',
+}
+
+@Entity()
+export class SquadPostModeration {
+  @PrimaryColumn({ type: 'text' })
+  id: string;
+
+  @Column({ type: 'text' })
+  sourceId: string;
+
+  @ManyToOne('Source', (source: Source) => source.id, {
+    lazy: true,
+    onDelete: 'CASCADE',
+  })
+  source: Promise<Source>;
+
+  @Column({
+    type: 'enum',
```

Comment:
Nice. We should actually start using this enum. How does it behave if we add more items to the enum? I assume we would have to run another migration? We might want to check if we want that maintenance overhead.

Reply:
@sshanzel Yes, I do think we would need to run another migration to update it:
```
await queryRunner.query(
 `CREATE TYPE "public"."squad_post_moderation_status_enum" AS ENUM('approved', 'rejected', 'pending')`,
);
```

and then:
```
await queryRunner.query(
 `CREATE TABLE "status" "public"."squad_post_moderation_status_enum" NOT NULL DEFAULT 'pending'
);
```

also trying to add a non specified value in TablePlus gives an error:
![Screenshot 2024-10-13 at 13 13 30](https://github.com/user-attachments/assets/04fec077-bf4e-429e-8a9f-53117de2436e)


Personally I think it's good practice to ensure only the desired values can be put into the database.

Reply:
Yeah, I agree, the stricter, the better. We can maybe raise this to the web guild as it has its pros and cons, and since I think this is probably the first usage of it. So that, moving forward, we can utilize the same type again if agreed upon.

Reply:
@AmarTrebinjac we usually validate enums on gql side. While psql enum is also good it requires a lot of effort to add new ones once the table grows. You can see on other examples how gql enum validation can be done.

Reply:
From safety perspective its the same since users always interact with entities through gql.